cli/azd/pkg/input/previewer.go

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package input
+
+import (
+	"strings"
+
+	tm "github.com/buger/goterm"
+)
+
+type previewer struct {
+	title        string
+	displayTitle string
+	footerLine   string
+	lines        int
+	prefix       string
+	output       []string
+}
+
+func NewPreviewer(lines int, prefix, title string) *previewer {
```

Comment:
Along similar lines, it would be really good to document each piece of state that `previewer` has and what they are used for.  We've been lax here because most of our objects don't have a ton of state that changes, instead they just have references to other services.  But for big components like this, it would be helpful to document what the state is.